pkg/commands/options/build.go

```diff
@@ -51,7 +51,7 @@ type BuildOptions struct {
 	ConcurrentBuilds     int
 	DisableOptimizations bool
 	SBOM                 string
-	Platform             string
+	Platforms            []string
```

Comment:
This breaks [Skaffold's `ko` integration](https://skaffold.dev/docs/pipeline-stages/builders/ko/), so I'd be cautious about this sort of change in general - as you mentioned in the PR.

However, in this particular case I'm ok with it, as I'm closely involved with Skaffold and I'm aware of this breaking change. It also means that I can remove this hoop from the Skaffold codebase: https://github.com/GoogleContainerTools/skaffold/blob/v1.35.1/pkg/skaffold/build/ko/builder.go#L53

We should flag the breaking change somehow to anyone else who might be embedding ko though - there could be private repos/tools out there that would be impacted.

A more cautious approach would be to retain the `Platform` field with a note that it has been deprecated, and use its value if `Platforms` is empty.

Reply:
This PR was partially the reason I wanted #554 -- if there's a change that would make the CLI better, like this one, but that might break users of the Go package in small and easy-to-workaround ways, I think we should do it.

If you'd like more time to stage the change, we can keep `Platform` as `Deprecated` and special-case it for one release.

As an alternative, it looks like you could just change your code to `Platforms: `.KoArtifact.Platforms,` when you bump your `ko` dep to include this change. That's pretty straightforward (and cleaner anyway), and I imagine most callers would have a similar diff. We should call out the breaking change in release notes, definitely.

Reply:
Yes, I agree. We don't need a phased approach in this case.